app/src/androidTest/java/com/android/streetworkapp/ui/Progression/ProgressionTest.kt

```diff
@@ -1,35 +1,147 @@
+    composeTestRule.onNodeWithTag("metricCardParksVisitedValue").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("metricCardFriendsAddedValue").assertIsDisplayed()
+  }
+
+  @Test
+  fun screenDisplaysCorrectElementsOnTypicalInputWithNonEmptyAchievements() {
+
+    val mockedProgression =
+        Progression(
+            progressionId = "prog123456",
+            uid = mockedUser.uid,
+            currentGoal = Ranks.SILVER.score,
+            eventsCreated = 9,
+            eventsJoined = 14,
+            achievements = listOf(MedalsAchievement.BRONZE.toString()))
+
+    val getProgressionOnSuccessSlot = slot<(Progression) -> Unit>()
+    every {
+      progressionRepository.getProgression(any(), capture(getProgressionOnSuccessSlot), any())
+    } answers { getProgressionOnSuccessSlot.captured(mockedProgression) }
+
+    composeTestRule.setContent {
+      ProgressScreen(navigationActions, userViewModel, progressionViewModel)
+    }
+
+    composeTestRule
+        .onNodeWithTag("percentageInsideCircularProgressBar")
+        .assertTextEquals(
+            "${(mockedUser.score / mockedProgression.currentGoal.toFloat() * 100).toInt()}%")
+    composeTestRule
+        .onNodeWithTag("scoreTextUnderCircularProgressBar")
+        .assertTextEquals("${mockedUser.score}/${mockedProgression.currentGoal}")
+
+    composeTestRule.onNodeWithTag("metricCardScoreValue").assertTextEquals("${mockedUser.score}")
+    // composeTestRule.onNodeWithTag("metricCardParksVisitedValue").assertTextEquals(INPUT_VALUE)
+    // TODO: commented out as it is not yet implemented
+    composeTestRule
+        .onNodeWithTag("metricCardFriendsAddedValue")
+        .assertTextEquals("${mockedUser.friends.size}")
+
+    composeTestRule.onNodeWithTag("emptyAchievementsText").assertIsNotDisplayed()
+
+    mockedProgression.achievements.forEachIndexed { index, achievementName
+      -> // only one type of achievement for now, but since we'll scale things later I make the test
+      // easily
+      // scalable
+      val achievement = enumValueOf<MedalsAchievement>(achievementName).achievement
+      val achievementItem = composeTestRule.onNodeWithTag("achievementItem${index}")
+    }
+  }
+
+  @Test
+  fun screenDisplaysEmptyAchievementsTextIfNoAchievements() {
+    val mockedProgression =
+        Progression(
+            progressionId = "prog123456",
+            uid = mockedUser.uid,
+            currentGoal = Ranks.BRONZE.score,
+            eventsCreated = 0,
+            eventsJoined = 0,
+            achievements = emptyList())
+
+    val getProgressionOnSuccessSlot = slot<(Progression) -> Unit>()
+    every {
+      progressionRepository.getProgression(any(), capture(getProgressionOnSuccessSlot), any())
+    } answers { getProgressionOnSuccessSlot.captured(mockedProgression) }
+
+    composeTestRule.setContent {
+      ProgressScreen(navigationActions, userViewModel, progressionViewModel)
+    }
```

Comment:
```
composeTestRule.setContent {
      ProgressScreen(navigationActions, userViewModel, progressionViewModel)
    }
```
set content is used in all of the test you can put it in the @before setup() function
code in line : 58, 66, 95, 141

Reply:
This is not relevant to this pr:
_Note: Ignore all files that are related to Progression, I had to merge the branch from pr https://github.com/SwEnt-Group8/Street-work-app/pull/93 into this one to integrate the end2end._
I'll take into account for https://github.com/SwEnt-Group8/Street-work-app/pull/93